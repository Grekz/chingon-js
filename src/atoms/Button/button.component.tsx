import styled from '@emotion/styled'
import * as React from 'react'
interface IButton {
  color?: string
}

type ButtonType = React.SFC<IButton>

const WrapperButton: ButtonType = styled.button`
  color: ${(props: IButton) => props.color};
  background: black;
`

const Button: ButtonType = props => <WrapperButton>{props}</WrapperButton>

export default Button
