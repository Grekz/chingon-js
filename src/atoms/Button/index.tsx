export { default, ButtonType } from './button.component'
