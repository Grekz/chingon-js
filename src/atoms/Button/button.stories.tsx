import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Button from '.'
import { ButtonTypes } from './button.component'

const { PRIMARY, SECONDARY, SUCCESS, WARNING, DANGER, INFO } = ButtonTypes

storiesOf('Atoms|Button', module).add('As Default', () => (
  <>
    <Button buttonType={PRIMARY}>Primary</Button>
    &nbsp;
    <Button buttonType={SECONDARY}>Secondary</Button>
    &nbsp;
    <Button buttonType={SUCCESS}>Success</Button>
    &nbsp;
    <Button buttonType={WARNING}>Warning</Button>
    &nbsp;
    <Button buttonType={DANGER}>Danger</Button>
    &nbsp;
    <Button buttonType={INFO}>Info</Button>
    &nbsp;
  </>
))
