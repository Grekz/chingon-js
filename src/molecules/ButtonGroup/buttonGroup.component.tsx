import styled from '@emotion/styled'
import * as React from 'react'

import { ButtonType } from '../../atoms/Button'

interface IButtonGroupProps {
  children: ButtonType[] | ButtonType
  role: string
}

export type ButtonGroupType = React.SFC<IButtonGroupProps>

const ButtonGroup = props => (
  <ButtonGroupWrapper role="group">{props.children}</ButtonGroupWrapper>
)

ButtonGroup.defaultProps = {}

const ButtonGroupWrapper: ButtonGroupType = styled.div`
  position: relative;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  > button:not(:last-child):not(.dropdown-toggle) {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
  > button:not(:first-child) {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
`

export default ButtonGroup
