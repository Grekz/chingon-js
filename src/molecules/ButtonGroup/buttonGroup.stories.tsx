import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Button from '../../atoms/Button'
import { ButtonTypes } from '../../atoms/Button/button.component'

import ButtonGroup from './'

const { PRIMARY } = ButtonTypes

storiesOf('Molecules|ButtonGroup', module).add('As Default', () => (
  <ButtonGroup>
    <Button buttonType={PRIMARY}>Left</Button>
    <Button buttonType={PRIMARY}>Center</Button>
    <Button buttonType={PRIMARY}>Right</Button>
  </ButtonGroup>
))
