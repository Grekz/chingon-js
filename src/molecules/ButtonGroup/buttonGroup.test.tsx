import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import ButtonGroup from '.'
import Button from '../../atoms/Button'
import { ButtonTypes } from '../../atoms/Button/button.component'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

const { PRIMARY } = ButtonTypes

describe('<ButtonGroup /> rendering', () => {
  it('should render ButtonGroup', () => {
    const wrapper = shallow(
      <ButtonGroup>
        <Button buttonType={PRIMARY}>Left</Button>
        <Button buttonType={PRIMARY}>Center</Button>
        <Button buttonType={PRIMARY}>Right</Button>
      </ButtonGroup>
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <ButtonGroup>
        <Button buttonType={PRIMARY}>Left</Button>
        <Button buttonType={PRIMARY}>Center</Button>
        <Button buttonType={PRIMARY}>Right</Button>
      </ButtonGroup>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
