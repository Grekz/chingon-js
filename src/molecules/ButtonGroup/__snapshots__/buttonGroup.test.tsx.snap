// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> rendering should match snapshot 1`] = `
<Styled(div)
  role="group"
>
  <Button
    buttonType="primary"
  >
    Left
  </Button>
  <Button
    buttonType="primary"
  >
    Center
  </Button>
  <Button
    buttonType="primary"
  >
    Right
  </Button>
</Styled(div)>
`;
